ouroboros-consensus/src/Ouroboros/Consensus/Util/SOP.hs

```diff
@@ -312,3 +323,54 @@ hizipWith4 ::
   -> h  f4 xs
   -> h  f5 xs
 hizipWith4 = hcizipWith4 (Proxy @Top)
+
+
+{-------------------------------------------------------------------------------
+  Tuples
```

Comment:
We should make sure that the types' names are consistent with the ones above this section.

Reply:
I think they are, aren't they?

Reply:
Sorry, I meant the type variable names. Eg we use either `a` or `x` to refer to these type variables.

Reply:
Fixed